data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java

```diff
@@ -140,6 +169,18 @@ public void encode(DataSchema schema) throws IOException
     // Initialize a new builder for the preferred encoding style
     _builder = _encodingStyle.newBuilderInstance(_writer);
 
+    // When counting column numbers, CompactPDLBuilder treats ',' as whitespace
```

Comment:
There is no guarantee that setEncodingStyle is called before encode. Seemed simpler to do this in one place otherwise would have to repeat this in the constructor too. This isn't a particularly expensive operation so seems ok to do it here. I'm ok moving it though if you feel strongly. Let me know.

Reply:
we can just call setEncodingStyle() from constructor.

Reply:
Done